scripts/actions/spells/black/death.lua

```diff
@@ -1,30 +1,70 @@
 -----------------------------------
 -- Spell: Death
--- Consumes all MP. Has a chance to knock out the target. If Death fails to knock out the target, it
--- will instead deal darkness damage. Ineffective against undead.
+-- Has a chance to insta-kill the target. Ineffective against undead or notorious monsters.
+-- (Player only) Consumes all MP no matter what.
+-- (Player only) If Death fails to knock out the target, it will instead deal darkness damage.
 -----------------------------------
 local spellObject = {}
 
 spellObject.onMagicCastingCheck = function(caster, target, spell)
     spell:setFlag(xi.magic.spellFlag.IGNORE_SHADOWS)
+
     return 0
 end
 
 spellObject.onSpellCast = function(caster, target, spell)
-    if
-        target:isUndead() or
-        target:hasStatusEffect(xi.effect.MAGIC_SHIELD) or
-        -- Todo: DeathRes has no place in the resistance functions so far..
-        math.random(1, 100) <= target:getMod(xi.mod.DEATHRES)
-    then
-        spell:setMsg(xi.msg.basic.MAGIC_NO_EFFECT)
-        return 0
-    end
+    -- Player spell.
+    if caster:isPC() or caster:isTrust() then
+        local instaDeath = false
+        local result     = 0
 
-    spell:setMsg(xi.msg.basic.FALL_TO_GROUND)
-    target:setHP(0)
+        -- Insta-death calculations.
+        if
+            not target:isUndead() and
+            not target:isNM() and
+            not target:hasStatusEffect(xi.effect.MAGIC_SHIELD)
+        then
+            local magicAcc     = xi.combat.magicHitRate.calculateActorMagicAccuracy(caster, target, xi.magic.spellGroup.BLACK, xi.skill.DARK_MAGIC, xi.element.DARK, 0, 0)
+            local magicEva     = xi.combat.magicHitRate.calculateTargetMagicEvasion(caster, target, xi.element.DARK, true, 0, 0)
+            local magicHitRate = utils.clamp(xi.combat.magicHitRate.calculateMagicHitRate(magicAcc, magicEva), 5, 30) -- Sources suggest a 30% max rate for players.
+            local resistRate   = xi.combat.magicHitRate.calculateResistRate(caster, target, xi.skill.DARK_MAGIC, xi.element.DARK, magicHitRate, 0)
 
-    return 0
+            if resistRate == 1 then
+                instaDeath = true
+            end
+        end
+
+        if instaDeath then
+            spell:setMsg(xi.msg.basic.FALL_TO_GROUND)
+            target:setHP(0)
+        else
+            spell:setMsg(xi.msg.basic.MAGIC_DMG)
+            result = xi.spells.damage.useDamageSpell(caster, target, spell)
+        end
+
+        -- Handle MP comsumption. It bypasses "conserve MP" and any other form of mp reduction.
+        caster:setMP(0)
+
+        return result
+
+    -- Not-player/trust spell.
+    else
+        local targetResistance = target:getMod(xi.mod.DEATHRES)
+        local randomRoll       = math.random(1, 100)
+
+        if targetResistance > 0 then
+            targetResistance = targetResistance + 5 -- Assuming it works as any other trait and gets an extra 5% at first level.
```

Comment:
> Assuming it works as any other trait 

Unfortunately it does not, there is no base 5% chance to escape it. Some non-spell based forms of it can be evaded by reraise (and will consume the reraise) but the cast just straight up kills you unless gear procs.